src/ui/ui_settings/ui_settings.js

```diff
@@ -0,0 +1,123 @@
+import { defaultsDeep } from 'lodash';
+import Bluebird from 'bluebird';
+
+import { getDefaultSettings } from './defaults';
+
+function hydrateUserSettings(user) {
+  return Object.keys(user).reduce(expand, {});
+  function expand(expanded, key) {
+    const userValue = user[key];
+    if (userValue !== null) {
+      expanded[key] = { userValue };
+    }
+    return expanded;
+  }
+}
+
+function assertRequest(req) {
+  if (
+    typeof req === 'object' &&
```

Comment:
You're right about both points. (styleguide shouldn't be out of date, but some of the rules in there are harder to follow since eslint doesn't enforce them.